relayer/chains/wasm/provider.go

```diff
@@ -570,41 +573,107 @@ func (p *Provider) fetchBlockMessages(ctx context.Context, heightInfo *types.Hei
 		wg           sync.WaitGroup
 		messages     coreTypes.ResultTxSearch
 		messagesChan = make(chan *coreTypes.ResultTxSearch)
-		errorChan    = make(chan error)
 	)
 
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
 	for _, event := range p.eventList {
 		wg.Add(1)
-		go func(wg *sync.WaitGroup, searchParam types.TxSearchParam, messagesChan chan *coreTypes.ResultTxSearch, errorChan chan error) {
+		go func(event sdkTypes.Event) {
 			defer wg.Done()
-			searchParam.Events = append(searchParam.Events, event)
-			res, err := p.client.TxSearch(ctx, searchParam)
-			if err != nil {
-				errorChan <- err
-				return
+
+			var (
+				localSearchParam = searchParam
```

Comment:
When any error occurs, the context is cancelled and returned and messages is processed. How will this be retried since we seem to be operating normally and just cancelling the context and nothing more on any errors. Can you shed some light here ?

Reply:
yeah, my bad.